userspace/libsinsp/container_engine/cri.cpp

```diff
@@ -243,7 +243,32 @@ bool cri::resolve(sinsp_threadinfo *tinfo, bool query_os_for_missing_info)
 			libsinsp_logger()->format(sinsp_logger::SEV_DEBUG,
 					"cri_async (%s): Starting synchronous lookup",
 					container_id.c_str());
+			// `lookup_sync` function directly invokes the container engine specific parser `parse`
 			done = m_async_source->lookup_sync(key, result);
+			if(!result.m_image.empty())
+			{
+				/*
+				* Only for synchronous lookup option (e.g. Falco's default is async not sync)
+				*
+				* Explicitly check for the most crucial retrieved value (`m_image`) to be present before enabling the 
+				* fast-track container add option. At this point, the container with only the cgroup (container id) was 
+				* already added to the cache. Therefore, we can proceed to call `replace_container`.
+				* 
+				* Bypassing the round-trip process:
+				* `source_callback` -> `notify_new_container` ->
+				* `container_to_sinsp_event(container_to_json(container_info), ...)` ->
+				* `parse_container_json_evt` -> `m_inspector->m_container_manager.add_container()`
+				*
+				* In `parse_container_json_evt`, we still re-add the container to support native 'container' events
```

Comment:
@mstemm if ok I would like to propose this approach and keep `parse_container_json_evt` that would trigger the callbacks as is as explained in this updated comment. From our previous discussion storing the container earlier as done here seemed acceptable.

Plus I have actually pre-tested this patch on some prod servers and seen amazing improvements. Not sure how much it will improve possible issues for other adopters, but at least it's some feedback.